Scroll snap (#469)
acss-io/atomizer: docs/tutorials/scroll.md

</div>

### [scroll-snap-align](/reference#scrollsnapalign) construct

The [scroll-snap-align](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-align) property specifies the box's snap position as an alignment of its snap area (as the alignment subject) within its snap container's snapport (as the alignment container). You can specify up to 2 values for this property, representing the block and inline axes, respectively. If you only give 1 value, that value will be used for both axes. Following example shows `Ssa(s)`, *scroll-snap-align: start*.

```html
<div class="Sb(s)">
    <div class="Ssa(s)">1</div>
    <div class="Ssa(c)">2</div>
    <div class="Ssa(e)">3</div>
</div>
```

<div class="Sb(s) Sst(x_m) Ovx(s) Ovy(h) W(200px) H(200px) D(f) Mt(20px)">
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(c) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(e) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
</div>

### [scroll-snap-type](/reference#scrollsnaptype) construct

> The [scroll-snap-type](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type) property specifies whether a scroll container is a scroll snap container, how strictly it snaps, and which axes are considered. If no strictness value is specified, proximity is assumed. Following example shows `Sst(x_p)`, *scroll-snap-type: x proximity*.

Reviewer: ```suggestion
The [scroll-snap-type](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type) property specifies whether a scroll container is a scroll snap container, how strictly it snaps, and which axis is considered. If no strictness value is specified, proximity is assumed. Following example shows `Sst(x_p)`, *scroll-snap-type: x proximity*.
```

---
section: docs
layout: docs
title: Scroll Snap
---

> These examples shows the various scroll snap properties supported by atomic.

Reviewer: ```suggestion
These examples shows the various scroll snap properties supported by Atomizer.
```

### `scroll-behavior` construct

> The scroll-behavior CSS property sets the behavior for a scrolling box when scrolling is triggered by the navigation, `CSSOM` scrolling APIs or `SCROLL-SNAP`.

Reviewer: I like linking to the MDN article for the user so they can link off to the docs pretty quickly. Let's do that for the other mentions below.

```suggestion
The [scroll-behavior](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior) CSS property sets the behavior for a scrolling box when scrolling is triggered by the navigation, `CSSOM` scrolling APIs or `SCROLL-SNAP`.
```

Why is `SCROLL-SNAP` in uppercase?

```html

> <div class="Sb(s)">

Reviewer: I like that the example is simple but the demo below has all the right classes

Author: I agree I will make all the example code only have the properties being discussed and have the html with all the required values, so is easer to explain the specific atomic class

    <div>1</div>
    <div>2</div>
    <div>3</div>
</div>
```

<div class="Sb(s) Sst(x_m) Ovx(s) Ovy(h) W(200px) H(200px) D(f) Mt(20px)">
    <div class="Ssa(s) Miw(200px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(s) Miw(200px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(s) Miw(200px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
</div>

### `scroll-snap-align` construct

You can specify up to 2 values for this property, representing the block and inline axes, respectively. If you only give 1 value, that value will be used for both axes.

```html
<div class="Sb(s)">
    <div class="Ssa(s)">1</div>
    <div class="Ssa(c)">2</div>
    <div class="Ssa(e)">3</div>
</div>
```

<div class="Sb(s) Sst(x_m) Ovx(s) Ovy(h) W(200px) H(200px) D(f) Mt(20px)">
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(c) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(e) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
</div>

### `scroll-snap-type` construct

> The `scroll-snap-type` property specifies whether a scroll container is a scroll snap container, how strictly it snaps, and which axes are considered. If no strictness value is specified, proximity is assumed.

Reviewer: Link `scroll-snap-type- to MDN

```html

> <div class="Sb(s) Sst(x_p)">

Reviewer: You should probably explain the `x_p` value in a paragraph above and perhaps link off to the MDN docs.

    <div>1</div>
    <div>2</div>
    <div>3</div>
</div>
```

<div class="Sb(s) Sst(x_p) Ovx(s) Ovy(h) W(160px) H(200px) D(f) Mt(20px)">
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
</div>

### `scroll-margin` && `scroll-margin-*` constructs

> The `scroll-margin` shorthand property sets scroll margin on all sides of an element at once, much like the `margin` property does for margin on an element. If all elements have the same spacing requirements, consider using `scroll-padding` on the parent container instead of `scroll-margin` because that affects spacing for all elements within the container.

Reviewer: Link `scroll-margin` to MDN.

Is `Smstart()` the appropriate class? Maybe that should be mentioned in the paragraph above.

```markdown
The [`scroll-margin`](link/to/mdn) shorthand property (`Smstart()`) sets scroll ...
```

Author: It's exactly like margins `Mstart` or Paddings `Pstart` `Pend` etc... but maybe we can make that clearer?

```html
<div class="Sb(s) Sst(x_m)">
    <div>1</div>
    <div class="Smstart(2em)">2</div>
    <div>3</div>
</div>
```

<div class="Sb(s) Sst(x_m) Ovx(s) Ovy(h) W(180px) H(200px) D(f) Gp(20px) Mt(20px)">
    <div class="Ssa(c) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(c) Smstart(2em) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(c) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
</div>

### `scroll-padding` & `scroll-padding-*`constructs

> The `scroll-padding` shorthand property sets scroll padding on all sides of an element at once, much like the `padding` property does for padding on an element.

Reviewer: Again link `scroll-padding` to MDN. Perhaps add the Atomizer class here too so its clear

```html
<div class="Sb(s) Sst(x_m) Spx(20px) Px(20px)">
    <div>1</div>
    <div>2</div>
    <div>3</div>
    ...
</div>
```

<div class="Sb(s) Sst(x_m) Ovx(s) Ovy(h) Maw(738px) H(200px) D(f) Gp(20px) Spx(20px) Px(20px) Bdc(#add8e6) Bdw(1px) Bds(s) Mt(20px)">
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">1</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">2</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">3</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">4</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">5</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#ccc)">6</div>
    <div class="Ssa(s) Miw(160px) H(200px) D(f) Ai(c) Jc(c) Fz(2em) Bgc(#add8e6) C(#ff)">7 </div>
</div>

### `scroll-snap-stop` construct

> The `scroll-snap-stop` CSS property defines whether or not the scroll container is allowed to "pass over" possible snap positions.

Reviewer: Link to MDN again
